samples/matter/common/src/persistent_storage/backends/persistent_storage_secure.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "../persistent_storage_common.h"
+#include "util/finite_map.h"
+
+#include <protected_storage.h>
+
+namespace Nrf
+{
+class PersistentStorageSecure {
+protected:
+	PSErrorCode _Init();
+	PSErrorCode _Store(PersistentStorageNode *node, const void *data, size_t dataSize);
+	PSErrorCode _Load(PersistentStorageNode *node, void *data, size_t dataMaxSize, size_t &outSize);
+	PSErrorCode _HasEntry(PersistentStorageNode *node);
+	PSErrorCode _Remove(PersistentStorageNode *node);
+
+	PSErrorCode _InitSecure();
+	PSErrorCode _StoreSecure(PersistentStorageNode *node, const void *data, size_t dataSize);
+	PSErrorCode _LoadSecure(PersistentStorageNode *node, void *data, size_t dataMaxSize, size_t &outSize);
+	PSErrorCode _HasEntrySecure(PersistentStorageNode *node);
+	PSErrorCode _RemoveSecure(PersistentStorageNode *node);
+
+private:
+	/* Needed to interface with FiniteMap. */
+	struct StringWrapper {
+		using Str = const char *;
+
+		StringWrapper() : mStr(nullptr) {}
+		StringWrapper(Str device) : mStr(device) {}
+
+		~StringWrapper() {}
+
+		/* Disable copy semantics and implement move semantics. */
+		StringWrapper(const StringWrapper &other) = delete;
+		StringWrapper &operator=(const StringWrapper &other) = delete;
+
+		StringWrapper(StringWrapper &&other)
+		{
+			mStr = other.mStr;
+			other.mStr = nullptr;
+		}
+
+		StringWrapper &operator=(StringWrapper &&other)
+		{
+			if (this != &other) {
+				this->~StringWrapper();
+				mStr = other.mStr;
+				other.mStr = nullptr;
+			}
+			return *this;
+		}
+
+		operator bool() const { return mStr; }
+		/* This will work also when comparing against a raw string. */
+		bool operator==(const StringWrapper &other) { return (strcmp(mStr, other.mStr) == 0); }
+
+		Str mStr;
+	};
```

Comment:
I think we can move it to the common `utils` directory since it looks like it may also be helpful in the other components in the future. We can move it to the finite_map.h as well.

Reply:
I'd personally opt for keeping it application specific unless we find the next use case for it.